src/main/java/org/truffleruby/language/TruffleBootNodes.java

```diff
@@ -88,52 +89,119 @@ public DynamicObject forceContext() {
         }
     }
 
-    @CoreMethod(names = "main", onSingleton = true, optional = 1)
+    @CoreMethod(names = "main", onSingleton = true)
     public abstract static class MainNode extends CoreMethodArrayArgumentsNode {
 
+        @Child private SnippetNode findSFileSnippetNode;
+        @Child private SnippetNode checkSyntaxSnippetNode;
```

Comment:
These could be in `@Cached` since there is a single specialization (and even if there was not since this node is executed once).